### DISCUSSION TITLE
Unable to call component method. Public method [$emit] not found on component

### DISCUSSION THREAD

**Discussion by theoreaves** [livewire/livewire#8435] [1. Help]
I just installed a fresh version of laravel (no starter kits).  I added tailwindcss, then added livewire 3.
All is working, I can do a livewire form submit with no issue, however, I get this error when I do a "$wire.emit" command:

Unable to call component method. Public method [$emit] not found on component

here is a flair for the full error:
https://flareapp.io/share/353gjgE5

Here is my app.blade:

<!DOCTYPE html>
<html lang="{{ str_replace('_', '-', app()->getLocale()) }}">
<head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">

    @livewireStyles

    @vite('resources/css/app.css')
    <title>{{ $title ?? 'Page Title' }}</title>
</head>
<body>
{{ $slot }}
@livewireScripts
</body>
</html>

and here is my component's blade:

<div>

    <div class="w-full p-4"
         x-data
    >
    <form wire:submit.prevent="submitForm">
        <x-form-top-bar> Client Setup </x-form-top-bar>
        <div class="col-span-8">
            <x-animated-error-container/>
        </div>
       <div class="grid grid-cols-8">
           <div class="col-span-4">
           <x-form-input title='Domain' name='domain_name' :required="true" :uc="false"
                         placeholder="your-site.tailorbuilt.software"
                         :disabled="false" />
           </div>
           <div class="col-span-2">
               <x-form-input title='Computer Name' name='computer_name' :required="false"
                             :uc="false"
                             placeholder="MyCADComputer"
                             :disabled="false" />
           </div>
           <div class="col-span-2">
               <x-form-input title='Mac Address' name='mac_address' :required="false"
                             placeholder="AAABBBCCCDDDEEE"
                             :uc="false"
                             :disabled="false" />
           </div>
           <div class="col-span-8">
               <x-form-checkbox title='Auto Start This Connection' name='auto_start_this_connection' :required="false"/>
           </div>
           <div class='col-span-8'>
               <x-form-bar>
                       <x-form-save :mode='$mode'/>
               </x-form-bar>
           </div>
       </div>
    </form>
        <div>
            <table class="w-full">
                <tr class="bg-gray-600 text-white sticky top-0">
                    <td>
                        Domain
                    </td>
                    <td>
                        Computer Name
                    </td>
                    <td>
                        MAC Address
                    </td>
                    <td>
                        &nbsp;
                    </td>
                </tr>
                @foreach($connections as $connection)
                   <tr class="{{ $loop->odd ? 'bg-gray-400' : 'bg-gray-500' }}"
                   >
                       <td>
                           {{ $connection->domain_name ?? '' }}
                       </td>
                       <td>
                           {{ $connection->computer_name ?? '' }}
                       </td>
                       <td>
                           {{ $connection->mac_address ?? '' }}
                       </td>
                       <td class="flex">
                           <div
                               class="cursor-pointer bg-blue-tb2 text-white hover:bg-blue-tb1 border-blue-tb1 rounded p-2"
{{--                               @click="$wire.emit('goTo', {{ $connection->id }})"--}}
                           wire:click="$emit('goTo', {{ $connection->id }})"
                           >GO</div>
                       </td>
                   </tr>
                @endforeach
            </table>
        </div>
    </div>

    </div>

I've tried wire:click and @click. Both give me the same error.

**Comment by joshhanley** [comment]
@theoreaves emit doesn't exist in V3, that is a V2 method name. It has been renamed to dispatch. See here for more details https://livewire.laravel.com/docs/upgrading#events

**Comment by theoreaves** [comment]
Thanks.
